tests/test_events_ops.py

```diff
@@ -0,0 +1,174 @@
+            "name" : "Partymad",
+            "location" : "Nairobu",
+            "description" : "here and 2",
+            "date": "2017-10-10",
+            "cost" : 2000
+        }
+        self.rsvp_ = {            
+            "eventid":1,            
+            "rsvp":"attending"
+        }
+
+        self.user = {
+            'email': 'test@kungu.com',
+            'username':'test',
+            'password':'hardpass'
+        }
+        self.login_details = {
+            'username' : 'test',
+            'password' : 'hardpass'
+        }
+
+        """Unit tests for events goes here"""    
+
+     # binds the app to the current context
+        with self.app.app_context():
+            # create all tables
+            db.create_all()
+
+    #register a user
+    def register_users(self):
+        user_details = json.dumps(self.user)
+        return self.client().post('api/v1/auth/register', data=user_details, content_type='application/json')
+
+    def open_with_auth(self, url, method, username, password):
+        return self.app.test_client().open(url,
+                   method = method,
+                   headers = {
+                        'Authorization': 'Basic ' + base64.b64encode(bytes(username + ":" + password, 'utf-8')).decode('utf-8') }
+    )
+
+    #login the newly created user
+    def user_login(self):        
+        return self.open_with_auth('/api/v1/auth/login', 'GET', 'test', 'hardpass')
+
+    #get the token from logged in user
+    def get_verfication_token(self):
+        
+        #register user 1st
+        self.register_users()
+
+        #login the user
+        res = self.user_login()
+        #from nose.tools import set_trace; set_trace()
+        #get the access token        
+        token = json.loads(res.data.decode('utf-8'))['token']
+        #from nose.tools import set_trace; set_trace()
+            
+        return token
+
+    def test_add_event(self):
+        token = self.get_verfication_token()   
+
+        res = self.client().post('/api/v1/events',
+                      headers = {'x-access-token' : token },
+                      data=json.dumps(self.event1), content_type='application/json')
+
+        self.assertEqual(res.status_code, 201)
+    
+    def test_add_event_has_location(self):
+        token = self.get_verfication_token()
+
+        res = self.client().post('/api/v1/events',
+              headers = {'x-access-token' : token },
+              data=json.dumps(self.event2), content_type='application/json')
+        self.assertEqual(res.status_code, 403)
+
+    def test_add_event_has_name(self):
+        token = self.get_verfication_token()
+
+        res = self.client().post('/api/v1/events', 
+            headers = {'x-access-token' : token },
+            data=json.dumps(self.event3), content_type='application/json')
+        self.assertEqual(res.status_code, 403)
+
+    #test api can get all events
+    def test_can_get_events(self):
```

Comment:
How sure It fetches all the events? it can return only one event instead of 20 and will still pass to be 200.